src/components/Orders/OrderForm.jsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from "react";
+import { addOrder, updateOrder } from "../../api/OrderService";
+
+const OrderForm = ({ order, onOrderAdded, onOrderUpdated }) => {
+	const [name, setName] = useState(order ? order.personalDetails.name : "");
+	const [email, setEmail] = useState(order ? order.personalDetails.email : "");
+	const [phone, setPhone] = useState(order ? order.personalDetails.phone : "");
+	const [street, setStreet] = useState(
+		order ? order.shippingDetails.street : ""
+	);
+	const [cardNumber, setCardNumber] = useState(
+		order ? order.shippingDetails.cardNumber : ""
+	);
+	const [isApproved, setIsApproved] = useState(
+		order ? order.isApproved : false
+	);
```

Comment:
We can go for single object state instead of handling them seprately

`const [formValues, setFormValues] = useState({
  name: order?.personalDetails?.name || "",
  email: order?.personalDetails?.email || "",
  phone: order?.personalDetails?.phone || "",
  street: order?.shippingDetails?.street || "",
  cardNumber: order?.shippingDetails?.cardNumber || "",
  isApproved: order?.isApproved || false,
});

const handleChange = (e) => {
  const { name, value } = e.target;
  setFormValues((prevValues) => ({
    ...prevValues,
    [name]: value,
  }));
};

const handleSubmit = async (e) => {
  e.preventDefault();
  // use formValues to create newOrder
};

<input
  type="text"
  name="name"
  value={formValues.name}
  onChange={handleChange}
  placeholder="Customer Name"
  required
/>
`